src/protocol.h

```diff
@@ -263,6 +269,8 @@ enum ServiceFlags : uint64_t {
     // NODE_WITNESS indicates that a node can be asked for blocks and transactions including
     // witness data.
     NODE_WITNESS = (1 << 3),
+    // NODE_ADDRv2 indicates send and receive support for ADDRv2 messages (BIP155).
+    NODE_ADDRv2 = (1 << 4),
```

Comment:
The signalling is done using a service flag because the service flags are gossiped and thus a node can choose to connect to `ADDRv2`-supporting nodes in the hopes of learning more e.g. Tor v3 addresses. For sure it is not going to learn any Tor v3, I2P or CJDNS addresses from nodes that do not support `ADDRv2`.

However, [BIP155](https://github.com/bitcoin/bips/blob/master/bip-0155.mediawiki#compatibility) mentions support should be signalled by protocol version bump. And [pull#16748 [WIP] Add support for addrv2 (BIP155)](https://github.com/bitcoin/bitcoin/pull/16748) implemented this via a dedicated new type of messages `sendaddrv2`.

I think that signalling via a service flag gives the most flexibility and thus I took the liberty to implement it that way, but I do not insist on it. I can change it to "protocol version bump", "dedicated new message type" or something else.